src/bosh_aws_cpi/lib/cloud/aws/cloud.rb

```diff
@@ -346,19 +346,27 @@ def snapshot_disk(disk_id, metadata)
         volume.attachments.each { |attachment| devices << attachment.device }
 
         name = ['deployment', 'job', 'index'].collect { |key| metadata[key] }
-        name << devices.first.split('/').last unless devices.empty?
+
+        unless devices.empty?
+          name << devices.first.split('/').last
+          metadata['device'] = devices.first
+        end
 
         snapshot = volume.create_snapshot(name.join('/'))
         logger.info("snapshot '#{snapshot.id}' of volume '#{disk_id}' created")
 
+        metadata.merge!({
+          'director' => metadata['director_name'],
+          'instance_index' => metadata['index'].to_s,
+          'instance_name' => metadata['job'] + '/' + metadata['instance_id'],
```

Comment:
Are these things we're changing on every CPI? Shouldn't they be changed at the higher level if they're changing? Seems like we're introducing inconsistencies if tags are being changed around on different IaaSes.